server-worker/src/test/scala/com/aliyun/emr/rss/service/deploy/cluster/ClusterReadWriteTest.scala

```diff
@@ -32,64 +32,72 @@ import com.aliyun.emr.rss.service.deploy.MiniClusterFeature
 class ClusterReadWriteTest extends MiniClusterFeature {
   @Test
   def testMiniCluster(): Unit = {
-    setUpMiniCluster()
-
-    val APP = "app-1"
-
-    val clientConf = new RssConf()
-    clientConf.set("rss.push.data.replicate", "true")
-    clientConf.set("rss.push.data.buffer.size", "256K")
-    val metaSystem = new LifecycleManager(APP, clientConf)
-    val shuffleClient = new ShuffleClientImpl(clientConf)
-    shuffleClient.setupMetaServiceRef(metaSystem.self)
-
-    val STR1 = RandomStringUtils.random(1024)
-    val DATA1 = STR1.getBytes(StandardCharsets.UTF_8)
-    val OFFSET1 = 0
-    val LENGTH1 = DATA1.length
-
-    val dataSize1 = shuffleClient.pushData(APP, 1, 0, 0, 0, DATA1, OFFSET1, LENGTH1, 1, 1)
-    logInfo(s"push data data size ${dataSize1}")
-
-    val STR2 = RandomStringUtils.random(32 * 1024)
-    val DATA2 = STR2.getBytes(StandardCharsets.UTF_8)
-    val OFFSET2 = 0
-    val LENGTH2 = DATA2.length
-    val dataSize2 = shuffleClient.pushData(APP, 1, 0, 0, 0, DATA2, OFFSET2, LENGTH2, 1, 1)
-    logInfo(s"push data data size ${dataSize2}")
-
-    val STR3 = RandomStringUtils.random(32 * 1024)
-    val DATA3 = STR3.getBytes(StandardCharsets.UTF_8)
-    val LENGTH3 = DATA3.length
-    val dataSize3 = shuffleClient.mergeData(APP, 1, 0, 0, 0, DATA3, 0, LENGTH3, 1, 1);
-
-    val STR4 = RandomStringUtils.random(16 * 1024)
-    val DATA4 = STR4.getBytes(StandardCharsets.UTF_8)
-    val LENGTH4 = DATA4.length
-    val dataSize4 = shuffleClient.mergeData(APP, 1, 0, 0, 0, DATA4, 0, LENGTH4, 1, 1);
-    shuffleClient.pushMergedData(APP, 1, 0, 0)
-    Thread.sleep(1000)
-
-    shuffleClient.mapperEnd(APP, 1, 0, 0, 1)
-
-    val inputStream = shuffleClient.readPartition(APP, 1, 0, 0)
-    val outputStream = new ByteArrayOutputStream()
-
-    var b = inputStream.read()
-    while (b != -1) {
-      outputStream.write(b)
-      b = inputStream.read()
+    CompressionCodec.values().foreach { codec =>
```

Comment:
`[ERROR]   ClusterReadWriteTest.testMiniCluster:35->$anonfun$testMiniCluster$1$adapted:35->$anonfun$testMiniCluster$1:80 Checksum not equal! expected: -48877264, actual: 1814018951`

``` java
clientConf.set("rss.client.compression.zstd.level", "-1")
```
when zstd level less than 0, test failed. So do our product cluster.

Reply:
Thanks for your feedback, created a related issue to fix it